### ISSUE TITLE
Merge advanced event tracking API

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#2092]
The advanced Analytics event tracking API from #1728 which is currently in the `feature/ga-advanced-tracking` should be merged into `develop`. Only the API itself should be included though, not any of the 3P plugin-specific code, which is more of a sample implementation and would preferably be implemented by the respective plugins.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Site Kit should offer a 3P API to register Analytics event tracking configurations.
    * The API should be configuration-based in a way that works in AMP and non-AMP. In other words, the actual configurations should not include any JavaScript code.
    * For AMP, it should be possible to transform the configuration into a JSON object of `amp-analytics` triggers. For non-AMP, a similar behavior should be achieved by injecting some minimal JavaScript code in the frontend that iterates through the configuration.
    * The JavaScript injected into the frontend for non-AMP should be _very_ backward-compatible, i.e. don't rely on recent browser features etc.
* No 3P plugin-specific code should be included in the Site Kit codebase.
* Test coverage for the new API should be expanded where possible.

## Implementation Brief

* Remove all 3P plugin-specific code from the `feature/ga-advanced-tracking` branch.
* Update branch against `develop`.
* Add extra test coverage.
* Complete and merge #2117.

## QA Brief

* Verify that the new 3P API satisfies the requirements to track Analytics events based on a configuration.

## Changelog entry

* N/A

**Comment by aaemnnosttv** [comment]
IB ✅

**Comment by eugene-manuilov** [comment]
QA :x:. The generated javascript file doesn't work on the frontend because it is built as a jsonp-based bundle.

**Comment by eugene-manuilov** [comment]
Created a PR against `master` branch that fixed javascript issues: https://github.com/google/site-kit-wp/pull/2152

**Comment by felixarntz** [comment]
After #2152, the code snippet added to the frontend works as expected. QA ✅
